server/service/src/sync/integrate_document.rs

```diff
@@ -0,0 +1,89 @@
+use repository::{
+    Document, DocumentRegistryFilter, DocumentRegistryRepository, DocumentRepository, EqualFilter,
+    RepositoryError, StorageConnection,
+};
+
+use crate::programs::{
+    encounter::{
+        encounter_updated::update_encounter_row, validate_misc::validate_encounter_schema,
+    },
+    patient::{patient_schema::SchemaPatient, patient_updated::update_patient_row},
+    program_enrolment::program_enrolment_updated::update_program_enrolment_row,
+    program_enrolment::program_schema::SchemaProgramEnrolment,
+};
+
+pub(crate) fn upsert_document(
+    con: &StorageConnection,
+    document: &Document,
+    is_sync_update: bool,
+) -> Result<(), RepositoryError> {
+    DocumentRepository::new(con).insert(document, is_sync_update)?;
+
+    let Some(registry) = DocumentRegistryRepository::new(con)
+        .query_by_filter(
+            DocumentRegistryFilter::new().document_type(EqualFilter::equal_to(&document.r#type)),
+        )?
+        .pop() else {
+        return Ok(());
```

Comment:
If we receive a document that is not in document registry, we quietly ignore it ? Shouldn't we have an error for this ? (it shoulnd't stop sync, but be captured in SyncBuffer)